### ISSUE TITLE
JetBooking v3.0.0 Beta

### ISSUE THREAD

**Issue by MjHead** [CrocoBlock/suggestions#7038]
Hi,

We are almost getting ready to release JetBooking v3.0.0 and we would like to ask you to help us test JetBooking v3.0.0 Beta.

You can download beta version from your Crocoblock account.

Please note that is a beta version, not stable release, so its not ready to use on production websites!

### General Points
At the moment, the JetBooking plugin and WooCommerce are integrated only at the payment stage, which creates certain inconveniences:

* You cannot book more than one property at a time.
* The booking form and the checkout form are separated, but you may still detect they have similar fields and would have to fill them in twice (given you haven’t set transferring fields from one form to another).
* A separate Product is created for processing booking orders, which is indexed by search engines and is available externally.

These problems can be solved by integrating JetBooking into WooCommerce directly, similar to the way the [WooCommerce Bookings plugin](https://woocommerce.com/products/woocommerce-bookings/) works. Basically, this integration would use WooCommerce Products as a default CPT for booking instances and embeds a calendar for picking dates directly into the add-to-cart form on the product’s single page. I.e., a user picks the dates on the product single page and adds the product to the cart. What happens next? On the checkout page, the order is placed, the payment is made, and a new booking is added to the Bookings List in the admin panel when checkout is completed.  

### How It’s Implemented. Settings

The settings will initially look the same as before except for one thing. Now, there will be an additional option to select the booking mode – [Screenshot](https://tppr.me/9UpG3). Two modes are available:

* Plain – this mode works like regular booking and includes basic JetBooking functionality; nothing new has been added to it.
* WooCommerce Based – a new booking type entirely based on WooCommerce. 

The booking mode settings will be shown depending on the enabled booking mode.

### How It’s Implemented. Product creation

As mentioned before, after you activate the WooCommerce Based mode, the entire booking system becomes connected to WooCommerce, and accordingly, the booking instance will be a Product. The main peculiarity here is that a regular WooCommerce-built store can work alongside this functionality, and there will be no conflicts between them. In this case, we added a new Product Type responsible for bookings, “JetBooking product” – https://tppr.me/XmnJU. You can switch to it in the Product data select like other product types.

Please note that the meta fields for setting up bookings appear only when you pick “JetBooking product,” not any other Product Type –[Screenshot](https://tppr.me/Go3BY). All tabs in the data panel have been hidden at this stage, as they are not of any functionality for product bookings; perhaps, we’ll change it in the future. In addition, we added a small description explaining how to configure everything to the main tab. Simply put, the booking settings for this product type will be completely the same as for a standard booking. Everything (from basic price, rates, and seasons to custom Datepicker configuration, individual product schedule, and unit settings) is done using meta boxes, and all booking-related meta boxes work accordingly. In turn, you can use the available WooCommerce fields; they will work, too. The product's standard price will be taken from the Price meta field.

Here are a few examples of settings and their combinations:
* https://tppr.me/IKAky
* https://tppr.me/88le
* https://tppr.me/fqx6b

In general, the booking product setup does not differ much from that of a regular booking instance. However, the settings fields will be displayed only in the corresponding product type.

### How It’s Implemented. Front End

After switching to the product’s front-end view, we can see a simple product. In the add-to-cart form, there will be a field for choosing the booking dates – https://tppr.me/uV0g8 – and the button text will be changed. 

If we go back a little on the archive page, JetBooking products will be in the same list as other Product Types – [Screenshot](https://tppr.me/fRfZ6). Instead of the “Add to Cart” button, there will be a “View Details” button. After clicking it, one will be redirected to the single booking page, where it’s possible to manipulate the product booking. 

On the single page, the booking button is not available – [Screenshot](https://tppr.me/dotoo). It will become available only after the user selects dates –  https://tppr.me/y95u8. You should also note that after picking the dates, a separate row will appear showing the total cost of the current booking as per the dates selected in the date picker. The price will, therefore, be calculated considering all the price settings configured in the plugin, i.e., price rates, weekend and seasonal prices (according to Elementor’s Dynamic Tags applicable to booking prices). 

There won’t be a quantity selection field both on the archive and single pages. Because we have pre-determined it to be one booking instance at all times, so there is no need to pick quantity. 

In case the dates weren’t picked, but the button somehow becomes available, and the user clicks to book a product, they will see a notice saying they need to select dates – https://tppr.me/oiLvj. Nothing will be added to the cart, though.

When adding a product to the cart, the plugin immediately creates a booking (you will find more details below) and allows for parallel booking (when two people have opened the same product and have the same available dates). As soon as one of the customers adds the dates to the cart, they become unavailable. However, they might not update for the second customer so fast in the live mode. In this case, when the second customer adds the product to the cart with already booked dates, they will receive an error that these dates are no longer available – https://tppr.me/eD6f8. The page will reload, and the calendar will further display the already booked dates. 

### How It’s Implemented. Bookings List

Now, let’s switch to the admin panel; the first thing to notice here is the Bookings List – https://tppr.me/dEjG6. You can immediately see that the entire table styling has been slightly changed; the action buttons are redesigned but preserve the same functionality.
You can also notice that the bookings have the same related order, which is one of the unique features of this mode. Even though there is one related order, the bookings are basically related to order items from this order – [Screenshot](https://tppr.me/cKneB).

Add New Booking pop-up. When WooCommerce Based mode is enabled, by default, there will be fields for creating an order inside the pop-up – https://tppr.me/NI7m2. These fields are required since the system is entirely based on WooCommerce functionality.

Edit Booking pop-up. If you change the booking item or booking date – https://tppr.me/8MDbd, you will see an additional control responsible for recalculating the order’s total price. 

The point is we already have a ready-made booking, and according to the order, we have to pay 2000, but it turned out that this booking item is unavailable for some reason. So, the owner offered to change the booking item to another one, which cost 3000 for the same dates. The owner does not turn on this option because he wants to apologize to the client for the inconvenience. As a result, we will spend our vacation in a better room for the same money. It can work the other way around, too. Upon the client's request, the owner recalculates the cost and activates this toggle; then, the price will be changed in the order and the booking item. The same principle applies to changing dates. But you can change everything together if needed.

### How It’s Implemented. WooCommerce Order

Regarding the order itself, we’ll start by saying that if you move the order to Trash like a post, then all related bookings’ statuses will be changed to “Cancelled” – https://tppr.me/PqH2V, https://tppr.me/uangS. If you delete the order permanently, all related bookings will also be deleted.

In the order itself, each item has its own booking date – https://tppr.me/qZ1P5 – similar to how it is done on the front end but slightly differently. Here, we can see the status of each item (because in the same order, booking items can be processed at different times) and a link to view all booking details. If you follow the link – https://tppr.me/ZsoIU – it will take you to a page with a list of all bookings and will open a pop-up with the corresponding booking details. 

Updating the order status – https://tppr.me/B0813 – will update the statuses of all related bookings – https://tppr.me/ka2sa.

In the order, you can issue a refund for the booking. It is possible to refund the amount both partially and wholly and the item itself. Different refund amount cases are presented from top to bottom – https://tppr.me/rEpos, https://tppr.me/CEwpP. Now, let’s see how refunds are shown in the booking – https://tppr.me/CEwpP – only the booking item to be fully refunded has its status changed to “Refunded.” It can be due to the fact that the owner decided to make it up to the client for some inconveniences and compensate them fully or partially. The booking, in turn, is not canceled, and the client gets to keep it.

**More info**
* https://youtu.be/0ZHVsGsh7Ko
* https://crocoblock.com/knowledge-base/jetbooking/how-to-use-booking-with-woocommerce/

**Please leave your feedback and any found bugs or issues in the comments below**

**Comment by Lonsdale201** [comment]
It is great to see how JetBooking is evolving. For me, Wp Booking System is the most preferred add-on, but more and more I feel that JetBooking is starting to reach a level that I am happy about, because I feel more comfortable working with your family of add-ons.

However, will there be a solution in the future (if say (not on a WC basis) but natively when you book, to allow the generated Unit system to be booked for multiple people with one booking?

It is ok to have a field with the number of persons, however as far as I know this cannot be directly linked to the Unit system. 

So for two and three people, double and triple bookings have to be made with the form. Is there any chance that there will be a solution for this in the future, do you think? Thanks for your answer in advance :)

**Comment by vivaldi007** [comment]
Got an error when trying to edit single product elementor template.
`WordPress has a built-in feature that detects when a plugin or theme causes a fatal error on your site, and notifies you with this automated email.

In this case, WordPress caught an error with one of your plugins, JetBooking.

First, visit your website and check for any visible issues. Next, visit the page where the error was caught  and check for any visible issues.

Please contact your host for assistance with investigating this issue further.

If your site appears broken and you can't access your dashboard normally, WordPress now has a special "recovery mode". This lets you safely login to your dashboard and investigate further.



To keep your site safe, this link will expire in 1 day. Don't worry about that, though: a new link will be emailed to you if the error occurs again after it expires.

When seeking help with this issue, you may be asked for some of the following information:
WordPress version 6.3
Active theme: Hello Elementor Child (version 1.0.1)
Current plugin: JetBooking (version 3.0.0-beta1)
PHP version 8.2.9



Error Details
=============
An error of type E_ERROR was caused in line 264 of the file /wp-content/plugins/jet-booking/includes/price.php. Error message: Uncaught Error: Call to a member function get_price() on null in /wp-content/plugins/jet-booking/includes/price.php:264
Stack trace:
#0 /wp-content/plugins/jet-booking/includes/price.php(204): JET_ABAF\Price->get_price_value()
#1 /wp-content/plugins/jet-booking/includes/price.php(143): JET_ABAF\Price->get_min_max_range()
#2 /wp-content/plugins/jet-booking/includes/price.php(114): JET_ABAF\Price->get_price()
#3 /wp-content/plugins/jet-booking/includes/elementor-integration/dynamic-tags/price-per-night.php(96): JET_ABAF\Price->get_price_for_display()
#4 /wp-content/plugins/elementor/core/dynamic-tags/tag.php(35): JET_ABAF\Elementor_Integration\Dynamic_Tags\Price_Per_Night->render()
#5 /wp-content/plugins/elementor/core/dynamic-tags/manager.php(216): Elementor\Core\DynamicTags\Tag->get_content()
#6 [internal function]: Elementor\Core\DynamicTags\Manager->get_tag_data_content()
#7 /wp-content/plugins/elementor/core/dynamic-tags/manager.php(109): call_user_func_array()
#8 /wp-content/plugins/elementor/core/dynamic-tags/manager.php(74): Elementor\Core\DynamicTags\Manager->parse_tag_text()
#9 [internal function]: Elementor\Core\DynamicTags\Manager->Elementor\Core\DynamicTags\{closure}()
#10 /wp-content/plugins/elementor/core/dynamic-tags/manager.php(73): preg_replace_callback()
#11 /wp-content/plugins/elementor/includes/controls/base-data.php(91): Elementor\Core\DynamicTags\Manager->parse_tags_text()
#12 /wp-content/plugins/elementor/includes/base/controls-stack.php(1251): Elementor\Base_Data_Control->parse_tags()
#13 /wp-content/plugins/elementor/includes/base/controls-stack.php(1102): Elementor\Controls_Stack->parse_dynamic_settings()
#14 /wp-content/plugins/elementor/includes/base/controls-stack.php(1187): Elementor\Controls_Stack->get_parsed_dynamic_settings()
#15 /wp-content/plugins/elementor/includes/widgets/heading.php(305): Elementor\Controls_Stack->get_settings_for_display()
#16 /wp-content/plugins/elementor/includes/base/controls-stack.php(2243): Elementor\Widget_Heading->render()
#17 /wp-content/plugins/elementor/includes/base/widget-base.php(609): Elementor\Controls_Stack->render_by_mode()
#18 /wp-content/plugins/elementor/includes/base/widget-base.php(733): Elementor\Widget_Base->render_content()
#19 /wp-content/plugins/elementor/includes/base/element-base.php(536): Elementor\Widget_Base->get_raw_data()
#20 /wp-content/plugins/elementor/core/base/document.php(977): Elementor\Element_Base->get_raw_data()
#21 /wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php(444): Elementor\Core\Base\Document->get_elements_raw_data()
#22 /wp-content/plugins/elementor/core/base/document.php(643): ElementorPro\Modules\ThemeBuilder\Documents\Theme_Document->get_elements_raw_data()
#23 /wp-content/plugins/elementor/modules/library/documents/library-document.php(66): Elementor\Core\Base\Document->get_initial_config()
#24 /wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php(666): Elementor\Modules\Library\Documents\Library_Document->get_initial_config()
#25 /wp-content/plugins/elementor/includes/base/controls-stack.php(996): ElementorPro\Modules\ThemeBuilder\Documents\Theme_Document->get_initial_config()
#26 /wp-content/plugins/elementor/core/editor/config-providers/editor-common-client-env.php(33): Elementor\Controls_Stack->get_config()
#27 /wp-content/plugins/elementor/core/editor/config-providers/editor-common-configs.php(169): Elementor\Core\Editor\Config_Providers\Editor_Common_Client_Env::get_client_env()
#28 /wp-content/plugins/elementor/core/editor/config-providers/editor-v1-config-provider.php(46): Elementor\Core\Editor\Config_Providers\Editor_Common_Configs::get_client_env()
#29 /wp-content/plugins/elementor/core/editor/editor-loader.php(57): Elementor\Core\Editor\Config_Providers\Editor_V1_Config_Provider->get_client_env()
#30 /wp-content/plugins/elementor/core/editor/editor.php(357): Elementor\Core\Editor\Editor_Loader->print_client_env()
#31 /wp-includes/class-wp-hook.php(310): Elementor\Core\Editor\Editor->enqueue_scripts()
#32 /wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters()
#33 /wp-includes/plugin.php(517): WP_Hook->do_action()
#34 /wp-includes/script-loader.php(2225): do_action()
#35 /wp-includes/class-wp-hook.php(310): wp_enqueue_scripts()
#36 /wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters()
#37 /wp-includes/plugin.php(517): WP_Hook->do_action()
#38 /wp-includes/general-template.php(3053): do_action()
#39 /wp-content/plugins/elementor/includes/editor-templates/editor-wrapper.php(31): wp_head()
#40 /wp-content/plugins/elementor/core/editor/editor-loader.php(135): include('...')
#41 /wp-content/plugins/elementor/core/editor/editor.php(162): Elementor\Core\Editor\Editor_Loader->print_root_template()
#42 /wp-includes/class-wp-hook.php(310): Elementor\Core\Editor\Editor->init()
#43 /wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters()
#44 /wp-includes/plugin.php(517): WP_Hook->do_action()
#45 /wp-admin/admin.php(419): do_action()
#46 /wp-admin/post.php(12): require_once('...')
#47 {main}
 thrown`

**Comment by UraraReika** [comment]
> Got an error when trying to edit single product elementor template. `WordPress has a built-in feature that detects when a plugin or theme causes a fatal error on your site, and notifies you with this automated email.
> 
> In this case, WordPress caught an error with one of your plugins, JetBooking.
> 
> First, visit your website and check for any visible issues. Next, visit the page where the error was caught and check for any visible issues.
> 
> Please contact your host for assistance with investigating this issue further.
> 
> If your site appears broken and you can't access your dashboard normally, WordPress now has a special "recovery mode". This lets you safely login to your dashboard and investigate further.
> 
> To keep your site safe, this link will expire in 1 day. Don't worry about that, though: a new link will be emailed to you if the error occurs again after it expires.
> 
> When seeking help with this issue, you may be asked for some of the following information: WordPress version 6.3 Active theme: Hello Elementor Child (version 1.0.1) Current plugin: JetBooking (version 3.0.0-beta1) PHP version 8.2.9
> 
> # Error Details
> An error of type E_ERROR was caused in line 264 of the file /wp-content/plugins/jet-booking/includes/price.php. Error message: Uncaught Error: Call to a member function get_price() on null in /wp-content/plugins/jet-booking/includes/price.php:264 Stack trace: #0 /wp-content/plugins/jet-booking/includes/price.php(204): JET_ABAF\Price->get_price_value() #1 /wp-content/plugins/jet-booking/includes/price.php(143): JET_ABAF\Price->get_min_max_range() #2 /wp-content/plugins/jet-booking/includes/price.php(114): JET_ABAF\Price->get_price() #3 /wp-content/plugins/jet-booking/includes/elementor-integration/dynamic-tags/price-per-night.php(96): JET_ABAF\Price->get_price_for_display() #4 /wp-content/plugins/elementor/core/dynamic-tags/tag.php(35): JET_ABAF\Elementor_Integration\Dynamic_Tags\Price_Per_Night->render() #5 /wp-content/plugins/elementor/core/dynamic-tags/manager.php(216): Elementor\Core\DynamicTags\Tag->get_content() #6 [internal function]: Elementor\Core\DynamicTags\Manager->get_tag_data_content() #7 /wp-content/plugins/elementor/core/dynamic-tags/manager.php(109): call_user_func_array() #8 /wp-content/plugins/elementor/core/dynamic-tags/manager.php(74): Elementor\Core\DynamicTags\Manager->parse_tag_text() #9 [internal function]: Elementor\Core\DynamicTags\Manager->Elementor\Core\DynamicTags{closure}() #10 /wp-content/plugins/elementor/core/dynamic-tags/manager.php(73): preg_replace_callback() #11 /wp-content/plugins/elementor/includes/controls/base-data.php(91): Elementor\Core\DynamicTags\Manager->parse_tags_text() #12 /wp-content/plugins/elementor/includes/base/controls-stack.php(1251): Elementor\Base_Data_Control->parse_tags() #13 /wp-content/plugins/elementor/includes/base/controls-stack.php(1102): Elementor\Controls_Stack->parse_dynamic_settings() #14 /wp-content/plugins/elementor/includes/base/controls-stack.php(1187): Elementor\Controls_Stack->get_parsed_dynamic_settings() #15 /wp-content/plugins/elementor/includes/widgets/heading.php(305): Elementor\Controls_Stack->get_settings_for_display() #16 /wp-content/plugins/elementor/includes/base/controls-stack.php(2243): Elementor\Widget_Heading->render() #17 /wp-content/plugins/elementor/includes/base/widget-base.php(609): Elementor\Controls_Stack->render_by_mode() #18 /wp-content/plugins/elementor/includes/base/widget-base.php(733): Elementor\Widget_Base->render_content() #19 /wp-content/plugins/elementor/includes/base/element-base.php(536): Elementor\Widget_Base->get_raw_data() #20 /wp-content/plugins/elementor/core/base/document.php(977): Elementor\Element_Base->get_raw_data() #21 /wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php(444): Elementor\Core\Base\Document->get_elements_raw_data() #22 /wp-content/plugins/elementor/core/base/document.php(643): ElementorPro\Modules\ThemeBuilder\Documents\Theme_Document->get_elements_raw_data() #23 /wp-content/plugins/elementor/modules/library/documents/library-document.php(66): Elementor\Core\Base\Document->get_initial_config() #24 /wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php(666): Elementor\Modules\Library\Documents\Library_Document->get_initial_config() #25 /wp-content/plugins/elementor/includes/base/controls-stack.php(996): ElementorPro\Modules\ThemeBuilder\Documents\Theme_Document->get_initial_config() #26 /wp-content/plugins/elementor/core/editor/config-providers/editor-common-client-env.php(33): Elementor\Controls_Stack->get_config() #27 /wp-content/plugins/elementor/core/editor/config-providers/editor-common-configs.php(169): Elementor\Core\Editor\Config_Providers\Editor_Common_Client_Env::get_client_env() #28 /wp-content/plugins/elementor/core/editor/config-providers/editor-v1-config-provider.php(46): Elementor\Core\Editor\Config_Providers\Editor_Common_Configs::get_client_env() #29 /wp-content/plugins/elementor/core/editor/editor-loader.php(57): Elementor\Core\Editor\Config_Providers\Editor_V1_Config_Provider->get_client_env() #30 /wp-content/plugins/elementor/core/editor/editor.php(357): Elementor\Core\Editor\Editor_Loader->print_client_env() #31 /wp-includes/class-wp-hook.php(310): Elementor\Core\Editor\Editor->enqueue_scripts() #32 /wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters() #33 /wp-includes/plugin.php(517): WP_Hook->do_action() #34 /wp-includes/script-loader.php(2225): do_action() #35 /wp-includes/class-wp-hook.php(310): wp_enqueue_scripts() #36 /wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters() #37 /wp-includes/plugin.php(517): WP_Hook->do_action() #38 /wp-includes/general-template.php(3053): do_action() #39 /wp-content/plugins/elementor/includes/editor-templates/editor-wrapper.php(31): wp_head() #40 /wp-content/plugins/elementor/core/editor/editor-loader.php(135): include('...') #41 /wp-content/plugins/elementor/core/editor/editor.php(162): Elementor\Core\Editor\Editor_Loader->print_root_template() #42 /wp-includes/class-wp-hook.php(310): Elementor\Core\Editor\Editor->init() #43 /wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters() #44 /wp-includes/plugin.php(517): WP_Hook->do_action() #45 /wp-admin/admin.php(419): do_action() #46 /wp-admin/post.php(12): require_once('...') #47 {main} thrown`

Referring to the error in your message, we tried to reproduce it using a JetBooking: Price per day/night dynamic tag to display the price (we used different dynamic tag settings during testing). Since you didn't specify what the single template was created with, we tried three options (JetWooBuilder Single Product Template, Elementor PRO Single Product Template or JetThemeCore Single Product template). However, we were unable to reproduce the error on any of the templates despite the different dynamic tag settings.

In this case we assume that this issue is related to your site. Please [contact](https://support.crocoblock.com/support/home/ ) our support team so that we can investigate this issue in more detail.

You could also use this template https://github.com/Crocoblock/suggestions/blob/master/.github/ISSUE_TEMPLATE/bug_report.md and describe your problem here in more detail.

**Comment by BrunoTimbo** [comment]
Hello!

I am using a JetFormBuilder form, before the Wooocommerce checkout form, to store extra reservation information such as guests' names and identification documents, that is, for each person number defined in the room, their name and number must be collected of the document.

I'm using a dynamic scheme with two TEXT FIELDS inside a REPEATER FIELD, to display according to the chosen number of people. ([https://prnt.sc/mbtDXXmixdma](https://prnt.sc/mbtDXXmixdma)), ([https://prnt.sc/GvPUkVUpFGPT](https://prnt.sc/GvPUkVUpFGPT)). The user is then taken to the woocommerce checkout after sending it.

But the word “ARRAY” is being displayed incorrectly in the place where such data stored by REPEATER should be displayed.

Error locations:

• “Your order” at woocommerce checkout ([https://prnt.sc/5uEfu0YnWYh6](https://prnt.sc/5uEfu0YnWYh6));
• “cart popup” and “cart page” ([https://prnt.sc/tLmUizn1tt-Z](https://prnt.sc/tLmUizn1tt-Z));
• “Booking details” on the thank you page ([https://prnt.sc/_dwhep5kxCFy](https://prnt.sc/_dwhep5kxCFy));
• “Booking Details” in JetBooking reservations ([https://prnt.sc/9vteGoML2TTs](https://prnt.sc/9vteGoML2TTs));
• “Order Details” in Woocommerce orders ([https://prnt.sc/XFP4-z3PCXbT](https://prnt.sc/XFP4-z3PCXbT)).

The same error persists when using the “WooCommerce Cart & Checkout Action” addon.

There is only no error when this data is stored in a CPT, as a separate reserve, with TEXT FIELDS inside a REPEATER.

I hope I helped. You guys are awesome! TKS!

**Comment by UraraReika** [comment]
> Hello!
> 
> I am using a JetFormBuilder form, before the Wooocommerce checkout form, to store extra reservation information such as guests' names and identification documents, that is, for each person number defined in the room, their name and number must be collected of the document.
> 
> I'm using a dynamic scheme with two TEXT FIELDS inside a REPEATER FIELD, to display according to the chosen number of people. (https://prnt.sc/mbtDXXmixdma), (https://prnt.sc/GvPUkVUpFGPT). The user is then taken to the woocommerce checkout after sending it.
> 
> But the word “ARRAY” is being displayed incorrectly in the place where such data stored by REPEATER should be displayed.
> 
> Error locations:
> 
> • “Your order” at woocommerce checkout (https://prnt.sc/5uEfu0YnWYh6); • “cart popup” and “cart page” (https://prnt.sc/tLmUizn1tt-Z); • “Booking details” on the thank you page (https://prnt.sc/_dwhep5kxCFy); • “Booking Details” in JetBooking reservations (https://prnt.sc/9vteGoML2TTs); • “Order Details” in Woocommerce orders (https://prnt.sc/XFP4-z3PCXbT).
> 
> The same error persists when using the “WooCommerce Cart & Checkout Action” addon.
> 
> There is only no error when this data is stored in a CPT, as a separate reserve, with TEXT FIELDS inside a REPEATER.
> 
> I hope I helped. You guys are awesome! TKS!

Thanks for the feedback.

This error not related to JetBooking 3.0.0 beta version so in this case you better to create new issue. But I checked your request and should say that you need to handle repeater field output in each case. Because in your case you try to output array as a string. Please [contact](https://support.crocoblock.com/support/home/ ) our support team they will help to solve your issue.

Thank you!

Thank you for your understanding.

**Comment by nick-simon** [comment]
Will there be the possibility for a variable price?  Previously you could use the form builder to create a fixed price + variable price based on a number field.  (i.e. Hotel room that is $150 a night plus $50 per person)  This does not seem possible with this update.

**Comment by UraraReika** [comment]
> Will there be the possibility for a variable price? Previously you could use the form builder to create a fixed price + variable price based on a number field. (i.e. Hotel room that is $150 a night plus $50 per person) This does not seem possible with this update.

Right now we complete only basic functionality and will develop it further adding more and more different features.

**Comment by Qubadi** [comment]
Hello

Thank you for the new Jetbooking and i like it.
We have send some request about to adding time/ hours to the Jetbooking, and its still not have been added. 

And the last one is, how can we add extra service to Jetbooking in Woocommerce product please?

Can we use Grouped Product?
Thank u
